test/integration/gardenlet/seed/seed/seed_test.go

```diff
@@ -393,6 +396,13 @@ var _ = Describe("Seed controller tests", func() {
 							deployment := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: "gardener-resource-manager", Namespace: testNamespace.Name}}
 							return testClient.Get(ctx, client.ObjectKeyFromObject(deployment), deployment)
 						}).Should(BeNotFoundError())
+
+						// There can be case when CRD is marked for deletion but still not gone from cluster. Because of this
+						// test in which CRD is created manually can flake.
```

Comment:
```suggestion
						// We should wait for the CRD to be deleted since it is a cluster-scoped resource so that we do not interfere
						// with other test cases.
```